Main branche

KazuGeoge opened the pull request.

KazuGeoge committed:
Codable保存用[すぐ閉じるメモとして保存]

質問用でしたが解決したので作業用ブランチのためのメモ用にする。作業用びんブランチをプッシュしたらこのブランチはCloseする

KazuGeoge committed:
ご指摘頂いた部分の修正

講義にてご指摘いただいた
・シングルトンはtokenのみにし、Requestは毎回生成する。
・protocolのEditCellの名前をCellsIdTypeに変更し、TweetViewControllerを入れた時の関数名はtweetViewControllerとする。
・GetAllMessageの全記事取得とCreateMessageでの投稿で来たレスポンスはContentInfoModelというModelを作りひとまとめにする。必要な時に各要素を取り出す。
・サーバからの情報をModeに集約する。
・レスポンスではnilを許容する形で ?? の形は使わない

URLSessionをStatic funcでまとめる部分はクロージャとTrailingClosureについてすぐに理解が出来なかったため、来週までに理解を深めて行こうと思います。

kazuteru commented on PlacticeNet_ios/ViewController/TextEditViewController.swift:
```diff
@@ -48,11 +48,12 @@ class TextEditViewController: UIViewController, UITextViewDelegate {
-                upLoadType?.putMessage(tweet: tweetString, textID: tweetID)
-                tableReloadDelegate?.reloadData()
-                dismiss(animated: true, completion: nil)
+                tableViewDataSouce.contentsInfoModel[indexPath!].contents = tweetString // nil無しの条件のためアンラップを許容
```
unwrapしないでif let 使うことはできます？

kazuteru commented on PlacticeNet_ios/ViewController/TextEditViewController.swift:
```diff
@@ -48,11 +48,12 @@ class TextEditViewController: UIViewController, UITextViewDelegate {
-                tableReloadDelegate?.reloadData()
-                dismiss(animated: true, completion: nil)
+                tableViewDataSouce.contentsInfoModel[indexPath!].contents = tweetString // nil無しの条件のためアンラップを許容
+                if let tweetID = tableViewDataSouce.contentsInfoModel[indexPath!].id {
```
idは大文字表記にするか小文字表記にするか決めたほうがよいかと
小文字表記で統一するならばtweetIdになると思います

kazuteru commented on PlacticeNet_ios/model/APIModel/ReadAPIData.swift:
```diff
@@ -0,0 +1,43 @@
+class ReadAPIData: GetAllType {
+    
+    var contents:[String] = []
+    var idIndex:[Int] = []
```
ここContentsInfoModelのarrayじゃダメですかね？

kazuteru commented on PlacticeNet_ios/model/APIModel/CreateAPIData.swift:
```diff
@@ -0,0 +1,47 @@
+class CreateAPIData: CreateType {
+    
+    var contents:[String] = []
+    var idIndex:[Int] = []
```
ここContentsInfoModelのarrayじゃダメですかね？

kazuteru commented on PlacticeNet_ios/model/ContentsInfoModel.swift:
```diff
@@ -0,0 +1,20 @@
+    var contents: String?
+    var id: Int?
+    
+    init(contents: String, id: Int) {
```
structだと変数すべてを受け取るinitは自動でつくられるのでこれは不要です

https://qiita.com/shtnkgm/items/8b7979fc84a3cc065238#%E6%A7%8B%E9%80%A0%E4%BD%93struct%E3%81%AE%E3%82%A4%E3%83%8B%E3%82%B7%E3%83%A3%E3%83%A9%E3%82%A4%E3%82%B6

kazuteru commented on PlacticeNet_ios/model/TokenModel.swift:
```diff
@@ -8,14 +8,11 @@
     
-    init(contents: String, token: String) {
-        self.contents = contents
+    init(token: String) {
```
structだと変数すべてを受け取るinitは自動でつくられるのでこれは不要です

https://qiita.com/shtnkgm/items/8b7979fc84a3cc065238#%E6%A7%8B%E9%80%A0%E4%BD%93struct%E3%81%AE%E3%82%A4%E3%83%8B%E3%82%B7%E3%83%A3%E3%83%A9%E3%82%A4%E3%82%B6

kazuteru commented on PlacticeNet_ios/model/APIModel/UPLoadAPIData.swift:
```diff
@@ -12,16 +12,15 @@ class UPLoadAPIData: UpLoadType{
-        // TODO: 以下他クラスとまとめられないか
-        URLSession.shared.dataTask(with: SingletonURLRequest.req) { (data, responce, error) in
-            
+        if let req =  SingletonURLRequest.upLoadMessage(textID: textID, editedContent: tweet) {
```
uploadは一単語なのでキャメルにならないですね

kazuteru commented on PlacticeNet_ios/model/APIModel/SendAPIData.swift:
```diff
@@ -0,0 +1,56 @@
+                    }
+                }
+            }
+                .resume()
```
インデントずれてますかね

KazuGeoge committed:
画像アップロード途中

画像ファイルを作成しData型にしてhttpリクエストに入れる。
画像のアップロードが上手くできない状況です。

KazuGeoge committed:
修正と機能、Viewの追加

アニメーション系
・カメラ関係のアイコンをフェードイン、フェードアウトにする
・セルの削除をフワッと消せるようにする　アニメーション
・投稿の文字がアニメーションで潰れるからボタンを広げても綺麗にする
・アニメーション時間の調整
UX系
・テーブルViewの選択したセルの後を消す

機能系
・投稿の文字次第でセルの高さを変える
・セルの高さを変える
・シェアの機能をつける
・セルを押すとその投稿の詳細を追加する
・画像を押したら詳細が出るようにする
UI系
・アイコンをTextViewの線に被せないようにして中に収める
・ユーザーIDの設置
・セル内のUIを全体的に調整

画像系
・画像のアップロードを可能に(CreateでMessageだけ投稿したらそのレスポンスを受け取るメソッドで画像をアップロードする)
・TableViewの画像　再読み込みしない
・tableViewの画像　読み込んだらそのセルに保存する
・TableViewの画像　チラつき防止
・TableViewの画像　投稿した時にずれるのを修正（サーバには正しくUploadされている）
・編集でも投稿でも画像のアップロードを可能に

を行いました！ご確認よろしくお願いします！

kazuteru commented on PlacticeNet_ios/ViewController/ImageViewController.swift:
```diff
@@ -0,0 +1,25 @@
+class ImageViewController: UIViewController {
+
+    @IBOutlet weak var imageView: UIImageView!
+    var tapedImageView: UIImageView?
```
tapped or pressed

kazuteru commented on PlacticeNet_ios/ViewController/TextEditViewController.swift:
```diff
@@ -57,9 +63,16 @@ class TextEditViewController: UIViewController, UITextViewDelegate, UIImagePicke
         if let tweetString = textView.text {
             if let idexPathRow = indexPath {
```
ここの`if let`は一つにまとめられるかと

kazuteru commented on PlacticeNet_ios/ViewController/TextEditViewController.swift:
```diff
@@ -10,40 +10,46 @@ import UIKit
     var editMessage: String?
     var indexPath: Int?
-    var uploadImageType: UploadImageType?
+    var addImage: UIImage?
```
動詞 + 名詞なので関数名風ですね
addedなど形容詞的に使う方がよいです

kazuteru commented on PlacticeNet_ios/ViewController/TweetDetailViewController.swift:
```diff
@@ -0,0 +1,94 @@
+
+class TweetDetailViewController: UIViewController {
+
+    @IBOutlet weak var user_Id: UILabel!
```
userId

kazuteru commented on PlacticeNet_ios/ViewController/TweetDetailViewController.swift:
```diff
@@ -0,0 +1,94 @@
+    @IBAction func sharedContentButton(_ sender: Any) {
+        
+        if let messageText = tweetContent.text {
+            if let thumbnailImage = imageView.image {
```
ここもまとめれそうですね

kazuteru commented on PlacticeNet_ios/ViewController/TweetViewController.swift:
```diff
@@ -17,41 +17,67 @@ protocol CreateType {
     @IBOutlet weak var tweetButtonTrailing: NSLayoutConstraint!
-    
+    @IBOutlet weak var tweetButtonWidth: NSLayoutConstraint!
     private var isFarstTouch = true
```
first

kazuteru commented on PlacticeNet_ios/ViewController/TweetViewController.swift:
```diff
@@ -17,41 +17,67 @@ protocol CreateType {
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        tableView.indexPathsForSelectedRows?.forEach {
```
forEach使ってくのはいいですね！

kazuteru commented on PlacticeNet_ios/ViewController/TweetViewController.swift:
```diff
@@ -106,11 +153,20 @@ class TweetViewController: UIViewController, UITextViewDelegate, CellsIdType, Ta
-        textView.alpha = 1
+        
+        // 上記で変更したサイズを１秒かけて再描画する
+        UIView.animate(withDuration: 1, animations: {
```
せっかくなので上のanimateの書き方と揃えましょう

kazuteru commented on PlacticeNet_ios/ViewController/TweetViewController.swift:
```diff
@@ -133,49 +197,75 @@ class TweetViewController: UIViewController, UITextViewDelegate, CellsIdType, Ta
     // 編集画面の変更を反映
-    func reloadData() {
+    func reloadData(textId: Int, image: UIImage?) {
+        if image != nil {
```
guardをうまく使うとネストが減りますよ

kazuteru commented on PlacticeNet_ios/cells/ImagePickerView.swift:
```diff
@@ -0,0 +1,65 @@
+    var toTweetViewImage: ToTweetViewImage?
+    
+    func startUpCamera(isFromTextEditView: Bool) -> UIImagePickerController? {
+        // カメラが利用可能か?
```
ここもguard文の方がよいですね

kazuteru commented on PlacticeNet_ios/cells/ImagePickerView.swift:
```diff
@@ -0,0 +1,65 @@
+}
+
+// UIImagePickerControllerDelegate
+extension ImagePickerView: UIImagePickerControllerDelegate, UINavigationControllerDelegate {
```
extensionの使い分けええですね！

kazuteru commented on PlacticeNet_ios/model/APIModel/ImageAPIModel.swift:
```diff
@@ -8,32 +8,50 @@
-        if let catPictureURL = URL(string: imageURL) {
-            let req = URLRequest(url: catPictureURL)
-            URLSession.shared.dataTask(with: req) { (data, responce, error) in
+        for contentInfo in messageInfo {
```
ここもforeachの方がシンプルかと

kazuteru commented on PlacticeNet_ios/model/APIModel/ImageAPIModel.swift:
```diff
@@ -8,32 +8,50 @@
+                            }
+                            
+                            // データが帰ってきたらメインスレッドで描画をする
+                            DispatchQueue.main.sync {
```
昨日話したとおりスコープを意識して修正してみましょう

kazuteru commented on PlacticeNet_ios/model/APIModel/ImageUploadAPIData.swift:
```diff
@@ -9,18 +9,19 @@
+    
+    func setImage(textId: Int, image: UIImage?) {
+        
+        if let uploadImage = image {
```
もっとネストを減らす方向にもってけそう

KazuGeoge committed:
codableの構造体でのUIImage生成

codableの構造体でのUIImage生成

KazuGeoge committed:
Merge pull request #3 from KazuGeoge/try-codable-image

codableの構造体でのUIImage生成

KazuGeoge committed:
Revert "修正と機能、Viewの追加"

KazuGeoge committed:
Revert "修正と機能、Viewの追加"

This reverts commit 20bb340285847f4865719d7186f5ecbfcc5b3dda.

KazuGeoge committed:
Revert "修正と機能、Viewの追加"

This reverts commit 20bb340285847f4865719d7186f5ecbfcc5b3dda.

KazuGeoge committed:
Revert "Revert "修正と機能、Viewの追加""

This reverts commit 45724e36f08be263c475b81030f618026768bb08.

KazuGeoge committed:
Revert "修正と機能、Viewの追加"

This reverts commit 20bb340285847f4865719d7186f5ecbfcc5b3dda.
